### PR TITLE
Feature branch dagger

### PR DESCRIPTION
**Pull request by Eduard-Kozachenko**

### CODE REVIEW

File: app/src/main/java/com/eduard/caloriecounter/dagger/module/CollectionModule.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.eduard.caloriecounter.presentation.presenter.CollectionContract;
+
+import dagger.Module;
+import dagger.Provides;
+
+@Module
+public class CollectionModule {
+    @ActivityScope
+    @Provides
+    public CollectionContract providePresenter(CollectionContract contract) {
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/dagger/module/CollectionModule.java]: Incorrect. See example: 
https://github.com/pipi3gunD/dogs-examples/blob/master/app/src/main/java/com/eduard/dogs/dogs/di/PresentationModule.java

File: app/src/main/java/com/eduard/caloriecounter/presentation/activity/NotesListActivity.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import com.eduard.caloriecounter.presentation.utils.NavigatorUtils;
+import com.eduard.caloriecounter.presentation.utils.RecyclerItemClickListener;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.recyclerview.widget.StaggeredGridLayoutManager;
+
+public class NotesListActivity extends AppCompatActivity implements View.OnClickListener, RecyclerItemClickListener.OnRecyclerViewItemClickListener, AppConstants {
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/activity/NotesListActivity.java]: Remove AppConstants. Constants are static variables that you should **import**, and **not implement** like interfaces.

File: app/src/main/java/com/eduard/caloriecounter/presentation/activity/NotesListActivity.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    private FloatingActionButton floatingActionButton;
+
+    private NoteRepository noteRepository;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_note_list);
+
+        noteRepository = new NoteRepository(getApplicationContext());
```
Comment (vladislav-v-rusakov) [inline app/src/main/java/com/eduard/caloriecounter/presentation/activity/NotesListActivity.java]: Incorrect. Read Clean Architecture and see example that you have done before. 
Presenter -> UseCase -> Repository

**Review by vladislav-v-rusakov — APPROVED** [review]

File: app/src/main/java/com/eduard/caloriecounter/presentation/activity/NotesListActivity.java (hunk 2 above)
Comment (Eduard-Kozachenko) [reply app/src/main/java/com/eduard/caloriecounter/presentation/activity/NotesListActivity.java]: ready

File: app/src/main/java/com/eduard/caloriecounter/dagger/module/CollectionModule.java (hunk 1 above)
Comment (Eduard-Kozachenko) [reply app/src/main/java/com/eduard/caloriecounter/dagger/module/CollectionModule.java]: ready

**Pull request merged**